/Users/liamtsy/cmpm17_venv/lib/python3.10/site-packages/torchvision/transforms/v2/_deprecated.py:42: UserWarning: The transform `ToTensor()` is deprecated and will be removed in a future release. Instead, please use `v2.Compose([v2.ToImage(), v2.ToDtype(torch.float32, scale=True)])`.Output is equivalent up to float precision.
  warnings.warn(
[33m[W 2025-03-14 16:14:15,290][0m Trial 0 failed with parameters: {'learning_rate': 0.0002277544254655752, 'dropout_rate': 0.598515254242866, 'conv1_channels': 28, 'conv2_channels': 42, 'conv3_channels': 114, 'fc1_units': 254, 'batch_size': 16, 'conv1_kernel_size': 5, 'conv1_padding_size': 2, 'pool_kernel_size': 2, 'pool_stride_size': 2, 'conv2_kernel_size': 3, 'conv2_padding_size': 1, 'conv3_kernel_size': 3, 'conv3_padding_size': 2, 'activation': 'ReLU', 'optimizer': 'Adam', 'model_variant': 'basic', 'weight_decay': 0.0008175559857454668} because of the following error: TypeError("CNN_Main.__init__() got an unexpected keyword argument 'transform'").[0m
Traceback (most recent call last):
  File "/Users/liamtsy/cmpm17_venv/lib/python3.10/site-packages/optuna/study/_optimize.py", line 197, in _run_trial
    value_or_values = func(trial)
  File "/Users/liamtsy/Desktop/Exchange_student/UCSC/Winter_2025/CMPM17-02/cmpm-17-final-project/main_optuna.py", line 273, in objective
    model = CNN_Main(num_classes=6,
TypeError: CNN_Main.__init__() got an unexpected keyword argument 'transform'
[33m[W 2025-03-14 16:14:15,291][0m Trial 0 failed with value None.[0m
Traceback (most recent call last):
  File "/Users/liamtsy/Desktop/Exchange_student/UCSC/Winter_2025/CMPM17-02/cmpm-17-final-project/main_optuna.py", line 469, in <module>
    study.optimize(objective, n_trials=50)
  File "/Users/liamtsy/cmpm17_venv/lib/python3.10/site-packages/optuna/study/study.py", line 475, in optimize
    _optimize(
  File "/Users/liamtsy/cmpm17_venv/lib/python3.10/site-packages/optuna/study/_optimize.py", line 63, in _optimize
    _optimize_sequential(
  File "/Users/liamtsy/cmpm17_venv/lib/python3.10/site-packages/optuna/study/_optimize.py", line 160, in _optimize_sequential
    frozen_trial = _run_trial(study, func, catch)
  File "/Users/liamtsy/cmpm17_venv/lib/python3.10/site-packages/optuna/study/_optimize.py", line 248, in _run_trial
    raise func_err
  File "/Users/liamtsy/cmpm17_venv/lib/python3.10/site-packages/optuna/study/_optimize.py", line 197, in _run_trial
    value_or_values = func(trial)
  File "/Users/liamtsy/Desktop/Exchange_student/UCSC/Winter_2025/CMPM17-02/cmpm-17-final-project/main_optuna.py", line 273, in objective
    model = CNN_Main(num_classes=6,
TypeError: CNN_Main.__init__() got an unexpected keyword argument 'transform'
